lib/legacy.sh

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
+
+JVM_COMMON_BUILDPACK_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && cd .. && pwd)"
+
+source "${JVM_COMMON_BUILDPACK_DIR}/lib/output.sh"
+
+# This buildpack did undergo significant changes in 2025. Some internal functions were removed that might be used
+# by other buildpacks that use this buildpack as a library. To make their experience less painful, we output an error
+# message about this change and exit the shell if one of those functions are being used.
+legacy::install_removed_function_handler() {
+	for removed_function_name in "${@}"; do
```

Comment:
(otherwise Bash leaks the loop's variable globally)

```suggestion
	local removed_function_name
	for removed_function_name in "${@}"; do
```